src/libstd/io/cursor.rs

```diff
@@ -15,16 +15,63 @@ use cmp;
 use io::{self, SeekFrom, Error, ErrorKind};
 use slice;
 
-/// A `Cursor` is a type which wraps a non-I/O object to provide a `Seek`
-/// implementation.
+/// A `Cursor` wraps another type and provides it with a [`Seek`][seek]
+/// implementation. This is often done to allow I/O operations on a non-I/O
+/// type.
 ///
-/// Cursors are typically used with memory buffer objects in order to allow
-/// `Seek`, `Read`, and `Write` implementations. For example, common cursor types
-/// include `Cursor<Vec<u8>>` and `Cursor<&[u8]>`.
+/// [seek]: trait.Seek.html
 ///
-/// Implementations of the I/O traits for `Cursor<T>` are currently not generic
-/// over `T` itself. Instead, specific implementations are provided for various
-/// in-memory buffer types like `Vec<u8>` and `&[u8]`.
+/// Cursors are typically used with in-memory buffers to allow them to
+/// implement `Read` and/or `Write`, allowing these buffers to be used
+/// anywhere you might use a reader or writer that does actual I/O.
+///
+/// The standard library implements some I/O traits on various types which
+/// are commonly used as a buffer, like `Cursor<Vec<u8>>` and `Cursor<&[u8]>`.
+///
+/// # Examples
+///
+/// We may want to write bytes to a [`TcpStream`][tcpstream] in our production
+/// code, but use an in-memory buffer in our tests. We can do this with
+/// `Cursor`:
```

Comment:
I think this example may want to be tweaked to use the seeking aspect of `Cursor`, because otherwise you don't actually need a cursor at all for this case (e.g. just using `Vec<u8>` in the tests is fine).

That being said it's a nice example to have somewhere :)